rna/rna-alignment/rna-alignment.wdl

```diff
@@ -0,0 +1,193 @@
+    Array[File] outputStarGenomeFiles = star_align.outputStarGenomeFiles
+    Array[File] outputStarpass1Files = star_align.outputStarpass1Files
+  }
+}
+
+task star_align {
+
+  Array[File] fastq_1
+  Array[File] fastq_2
+  String sample_name
+  File star_index
+
+  Int star_vm_disk_size_gb
+  String star_vm_memory
+  String star_docker
+  String runtime_zones
+  Int preemptible_tries
+  Float star_timeout_hours
+
+  command<<<
+    set -o errexit
+    set -o nounset
+    set -o pipefail
+    set -o xtrace
+
+    # For each FASTQ1 file, extract fields to build up a list
+    # of readgroup tags.
+    RGTAGLIST=""
+    for FILE in ${sep=" " fastq_1}; do
+      SAMPLE_NAME="$(basename $FILE | cut -d. -f1)"
+
+      LINE="$(head -n 1 < <(zcat $FILE))"
+      FLOWCELL="$(echo $LINE | cut -d: -f3)"
+      LANE="$(echo $LINE | cut -d: -f4)"
+      INDEX="$(echo $LINE | cut -d: -f10)"
+
+      if [[ -z $LINE ]]; then
+        2>&1 "ERROR: Unable to parse first line from $FILE"
+        exit 1
+      fi
+      if [[ -z $FLOWCELL ]]; then
+        2>&1 "ERROR: Unable to parse flowcell from $LINE"
+        exit 1
+      fi
+      if [[ -z $LANE ]]; then
+        2>&1 "ERROR: Unable to parse lane from $LINE"
+        exit 1
+      fi
+      if [[ -z $INDEX ]]; then
+        2>&1 "ERROR: Unable to parse index from $LINE"
+        exit 1
+      fi
+
+      RGTAG="ID:$FLOWCELL.$INDEX.$LANE PL:ILLUMINA SM:$SAMPLE_NAME "
+      RGTAGLIST="$RGTAGLIST , $RGTAG"
+    done
+    RGTAGLIST="$(echo $RGTAGLIST | sed 's/^, //')"
+
+    # Untar the STAR index
+    mkdir star_index
+    tar xfz "${star_index}" --directory star_index --strip-components=1
+
+    mkdir output_dir
+
+    # Run STAR --runMode alignReads
+    # We run using timeout to prevent long-running workflows
+    # from wasteful preemption loops.
+    timeout "${star_timeout_hours}"h STAR \
+      --genomeDir star_index \
+      --runMode alignReads \
+      --twopassMode Basic \
+      --outFileNamePrefix output_dir/${sample_name}. \
+      --readFilesCommand zcat \
+      --readFilesIn ${sep=',' fastq_1} ${sep=',' fastq_2} \
+      --outSAMtype BAM Unsorted \
+      --outFilterType BySJout \
+      --outFilterMultimapNmax 20 \
+      --outFilterMismatchNmax 999 \
+      --outFilterMismatchNoverLmax 0.1 \
+      --alignIntronMax 1000000 \
+      --alignMatesGapMax 1000000 \
+      --alignSJoverhangMin 8 \
+      --alignSJDBoverhangMin 1 \
+      --chimOutType Junctions \
+      --chimSegmentMin 15 \
+      --chimJunctionOverhangMin 15 \
+      --runThreadN 16 \
+      --outSAMstrandField intronMotif \
+      --outSAMunmapped Within \
+      --outSAMattrRGline $RGTAGLIST
+  >>>
+
+  runtime {
+    docker: star_docker
+
+    disks: "local-disk " + star_vm_disk_size_gb + " HDD"
+
+    memory: star_vm_memory
+    cpu: 16
+    preemptible: preemptible_tries
+    zones: runtime_zones
+  }
+  output {
+    # 6 outputs produced:
+    #   - <sample-id>.Aligned.sortedByCoord.out.bam
```

Comment:
This version of the STAR workflow uses `--outSAMtype BAM Unsorted`.
The name of the BAM should reflect that.